src/meltano/core/plugin_discovery_service.py

```diff
@@ -482,7 +482,8 @@ def find_related_plugin_refs(
 
         runner_ref = target_plugin.runner
         if runner_ref:
-            related_plugin_refs.append(runner_ref)
+            if runner_ref.type in plugin_types:
+                related_plugin_refs.append(runner_ref)
```

Comment:
How about something like

```python
             if runner_ref and runner_ref.type in plugin_types:
                 related_plugin_refs.append(runner_ref)
```

Reply:
Did this syntax change to be supported recently? I expect that in cases where `runner_ref = None` the `runner_ref.type` check will raise an `AttributeError`, which is why I have nested the `if` statements 🤔

Reply:
@kgpayne I don't know how recently the syntax is supported but this works in Python 3.7 😅 

```python
>>> def process_obj(obj):
    if obj and obj.x == 1:
        print(obj.x)
    else:
        print("It's none")
... 
>>> class MyClass:
    x = 1
>>> 
>>> process_obj(MyClass())
1
>>> process_obj(None)
It's none
```

I think everything after the `and` is lazily evaluated. So only if the first statement is truthy, the second one is processed.